actors/builtin/miner/miner_actor.go

```diff
@@ -1723,9 +1734,11 @@ func handleProvingDeadline(rt Runtime) {
 			penaltyTarget := PledgePenaltyForUndeclaredFault(epochReward, pwrTotal.QualityAdjPower, penalizePowerTotal)
 			// Subtract the "ongoing" fault fee from the amount charged now, since it will be added on just below.
 			penaltyTarget = big.Sub(penaltyTarget, PledgePenaltyForDeclaredFault(epochReward, pwrTotal.QualityAdjPower, penalizePowerTotal))
-			penalty, err := st.UnlockUnvestedFunds(store, currEpoch, penaltyTarget)
+			penaltyFromVesting, penaltyFromBalance, err := st.PenalizeFundsInPriorityOrder(store, currEpoch, penaltyTarget, availableBalance)
 			builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to unlock penalty")
-			penaltyTotal = big.Add(penaltyTotal, penalty)
+			availableBalance = big.Sub(availableBalance, penaltyFromBalance)
+			penaltyTotal = big.Sum(penaltyTotal, penaltyFromVesting, penaltyFromBalance)
+			pledgeDelta = big.Sum(pledgeDelta, penaltyFromVesting.Neg())
```

Comment:
Nit
```suggestion
			pledgeDelta = big.Sub(pledgeDelta, penaltyFromVesting)
```
?

Ok if you prefer consistently adding tho